zooniverse/zoo-ml-subject-assistant
# Zooniverse Authentication (login/logout)

## PR Overview

Closes #12 

This PR adds basic Zooniverse Custom Front End authentication to the app.
- Users can login and logout of the app.
- this only works on the non-production environments (i.e. you need to login using a Zooniverse staging user account)
- ⚠️ The login/logout functionality uses the `oauth` module from the ol' [Panoptes JavaScript Client](https://github.com/zooniverse/panoptes-javascript-client/)
  - I had heard that we were moving away from oAuth (?) but after looking at how it's done in the `front-end-monorepo`'s `lib-classifier` (I couldn't grok the auth architecture for `app-project`), it seems that oAuth is still being used? I welcome any corrections to this perception.

External Dependencies:
- Panoptes oAuth Application setup. I've done this for `staging`, but haven't bothered with `production` yet.

Testing:
- Run the app via `npm start`
- Open `http://localhost:3000` (or `http://local.zooniverse.org:3000` if you've set it up your hosts) in your browser
- There's a login button. Click in.
- You should be directed to the Panoptes (staging) login page.
- Login. You should be redirected to the app, and your username should be displayed on the app.

### Status

Ready for review
